bundles/org.openhab.core.model.script/src/org/eclipse/smarthome/model/script/lib/NumberExtensions.java

```diff
@@ -89,7 +88,13 @@ public static BigDecimal operator_multiply(Number x, Number y) {
     public static BigDecimal operator_divide(Number x, Number y) {
         BigDecimal xValue = numberToBigDecimal(x);
         BigDecimal yValue = numberToBigDecimal(y);
-        return xValue.divide(yValue, 8, RoundingMode.HALF_UP);
+        if (xValue == null) {
+            return NULL_DEFINITION;
+        } else if (yValue == null) {
+            return NULL_DEFINITION;

```

Comment:
We could throw an `java.lang.ArithmeticException` in this case too?

Reply:
Why do we need to "mask" a `null` reference at all?
Why is `x / null` and `null / x` equal to zero?

Reply:
Throw an IAE then? If we change that we have to touch other methods too.

Reply:
@maggu2810 Wdyt we should do here? Did you see this comment:

https://github.com/openhab/openhab-core/blob/b9ac4cf1f5f05535878909436ec160e4b495ef06/bundles/org.openhab.core.model.script/src/org/eclipse/smarthome/model/script/lib/NumberExtensions.java#L33-L39

Reply:
> but we can use the null pointer and throws an NPE when a null value is used

So, doesn't it state that in this case it is absolutely valid to call a NPE as you are using a null value in a division.